### DISCUSSION TITLE
How does Import-TDBuild.ps1 work?

### DISCUSSION THREAD

**Discussion by Mysterious-Dev** [microsoft/TemplateStudio#4635] [Q&A] [answered]
Hey. I wanted to know, how Import-TDBuild.ps1 script work exactly ? Where we put TDBuild.tar.gz, if it's needed to use parameters for script, etc... Because I want to do a PR for french translation, and I don't really understand how this script work :/.

Thanks in advance ^^.

**Comment by mikebattista** [comment, marked as answer]
It's used in conjunction with an internal build pipeline that localizes resources on the backend. If you're proposing manual changes, then this script is not relevant.

The files that are onboarded to that pipeline are below.

https://github.com/microsoft/TemplateStudio/blob/39bb5c14efad5e3f00853d0886926d24646f9dfe/_build/pipelines/localization.yml#L37-L41

If you're looking to propose manual changes, I would focus on the files that aren't onboarded to that pipeline like the .md files. Though if you see issues with other files, let us know.

**Reply by Mysterious-Dev** [reply]
Okay, so if I want to localize description of feature templates, i don't need to use this script ? Because I imagine that it is not enough to simply create files with .fr-Fr.json or .fr-Fr.md.

**Reply by mikebattista** [reply]
You don't need the script, no, but if you try to manually localize files that are managed by the script, the script is going to overwrite any changes you make the next time we run it.

The Markdown description files aren't managed by the script so those would just be manual check-ins.

**Reply by Mysterious-Dev** [reply]
Okay, and for the script, it is used to send translatable resources to a site like Crowdin or for something else?

**Reply by mikebattista** [reply]
The pipeline sends English resources to an internal service for localization. The script imports the localized files into the repo.

**Reply by Mysterious-Dev** [reply]
The internal service is accessible for contributors (like a Crowdin) ?

**Reply by mikebattista** [reply]
No.

**Reply by mikebattista** [reply]
I just imported the latest localized strings with https://github.com/microsoft/TemplateStudio/issues/4645.
